Error 500 returned when can't retrieve data from coingecko.

**Describe the bug**
The service return a 500 Error when can't retrieve data from coingecko. 

**To Reproduce**
Taked from logs. 

**Expected behavior**
Maybe could be better return service not available 503. 

**Additional context**
```
2022-07-19 10:39:41,111 [ERROR] [MainProcess] Internal Server Error: /api/v1/tokens/0x0000000000000000000000000000000000000000/prices/usd/
Traceback (most recent call last):  File "/usr/local/lib/python3.10/site-packages/urllib3/connectionpool.py", line 703, in urlopen    httplib_response = self._make_request(  File "/usr/local/lib/python3.10/site-packages/urllib3/connectionpool.py", line 449, in _make_request    six.raise_from(e, None)  File "<string>", line 3, in raise_from  File "/usr/local/lib/python3.10/site-packages/urllib3/connectionpool.py", line 444, in _make_request
    httplib_response = conn.getresponse()
  File "/usr/local/lib/python3.10/http/client.py", line 1374, in getresponse
    response.begin()
  File "/usr/local/lib/python3.10/http/client.py", line 318, in begin
    version, status, reason = self._read_status()
  File "/usr/local/lib/python3.10/http/client.py", line 287, in _read_status
    raise RemoteDisconnected("Remote end closed connection without"
http.client.RemoteDisconnected: Remote end closed connection without response

During handling of the above exception, another exception occurred:

Traceback (most recent call last):  File "/usr/local/lib/python3.10/site-packages/requests/adapters.py", line 489, in send    resp = conn.urlopen(  File "/usr/local/lib/python3.10/site-packages/urllib3/connectionpool.py", line 787, in urlopen    retries = retries.increment(  File "/usr/local/lib/python3.10/site-packages/urllib3/util/retry.py", line 550, in increment    raise six.reraise(type(error), error, _stacktrace)  File "/usr/local/lib/python3.10/site-packages/urllib3/packages/six.py", line 769, in reraise    raise value.with_traceback(tb)  File "/usr/local/lib/python3.10/site-packages/urllib3/connectionpool.py", line 703, in urlopen    httplib_response = self._make_request(  File "/usr/local/lib/python3.10/site-packages/urllib3/connectionpool.py", line 449, in _make_request    six.raise_from(e, None)  File "<string>", line 3, in raise_from  File "/usr/local/lib/python3.10/site-packages/urllib3/connectionpool.py", line 444, in _make_request
    httplib_response = conn.getresponse()
  File "/usr/local/lib/python3.10/http/client.py", line 1374, in getresponse
    response.begin()
  File "/usr/local/lib/python3.10/http/client.py", line 318, in begin
    version, status, reason = self._read_status()
  File "/usr/local/lib/python3.10/http/client.py", line 287, in _read_status
    raise RemoteDisconnected("Remote end closed connection without"
urllib3.exceptions.ProtocolError: ('Connection aborted.', RemoteDisconnected('Remote end closed connection without response'))
During handling of the above exception, another exception occurred:

Traceback (most recent call last):  File "/app/safe_transaction_service/tokens/clients/coingecko_client.py", line 57, in _do_request    response = self.http_session.get(url, timeout=10)  File "/usr/local/lib/python3.10/site-packages/requests/sessions.py", line 600, in get    return self.request("GET", url, **kwargs)  File "/usr/local/lib/python3.10/site-packages/requests/sessions.py", line 587, in request    resp = self.send(prep, **send_kwargs)  File "/usr/local/lib/python3.10/site-packages/requests/sessions.py", line 701, in send    r = adapter.send(request, **kwargs)  File "/usr/local/lib/python3.10/site-packages/requests/adapters.py", line 547, in send    raise ConnectionError(err, request=request)requests.exceptions.ConnectionError: ('Connection aborted.', RemoteDisconnected('Remote end closed connection without response'))

The above exception was the direct cause of the following exception:

Traceback (most recent call last):  File "/app/safe_transaction_service/tokens/clients/coingecko_client.py", line 69, in _get_price    result = self._do_request(url)  File "/app/safe_transaction_service/tokens/clients/coingecko_client.py", line 65, in _do_request    raise CoingeckoRequestError from e
safe_transaction_service.tokens.clients.exceptions.CoingeckoRequestError

The above exception was the direct cause of the following exception:

Traceback (most recent call last):  File "/usr/local/lib/python3.10/site-packages/django/core/handlers/exception.py", line 55, in inner    response = get_response(request)  File "/usr/local/lib/python3.10/site-packages/django/core/handlers/base.py", line 197, in _get_response    response = wrapped_callback(request, *callback_args, **callback_kwargs)  File "/usr/local/lib/python3.10/site-packages/django/views/decorators/csrf.py", line 54, in wrapped_view    return view_func(*args, **kwargs)  File "/usr/local/lib/python3.10/site-packages/django/views/generic/base.py", line 84, in view    return self.dispatch(request, *args, **kwargs)  File "/usr/local/lib/python3.10/site-packages/rest_framework/views.py", line 509, in dispatch    response = self.handle_exception(exc)  File "/usr/local/lib/python3.10/site-packages/rest_framework/views.py", line 469, in handle_exception    self.raise_uncaught_exception(exc)  File "/usr/local/lib/python3.10/site-packages/rest_framework/views.py", line 480, in raise_uncaught_exception    raise exc  File "/usr/local/lib/python3.10/site-packages/rest_framework/views.py", line 506, in dispatch    response = handler(request, *args, **kwargs)  File "/usr/local/lib/python3.10/site-packages/django/utils/decorators.py", line 46, in _wrapper    return bound_method(*args, **kwargs)  File "/usr/local/lib/python3.10/site-packages/django/utils/decorators.py", line 133, in _wrapped_view    response = view_func(request, *args, **kwargs)  File "/app/safe_transaction_service/tokens/views.py", line 86, in get    "fiat_price": str(price_service.get_native_coin_usd_price()),  File "/usr/local/lib/python3.10/site-packages/cachetools/__init__.py", line 700, in wrapper    v = method(self, *args, **kwargs)  File "/usr/local/lib/python3.10/site-packages/cache_memoize/__init__.py", line 137, in inner    result = func(*args, **kwargs)  File "/app/safe_transaction_service/tokens/services/price_service.py", line 220, in get_native_coin_usd_price    return self.get_aurora_usd_price()  File "/app/safe_transaction_service/tokens/services/price_service.py", line 156, in get_aurora_usd_price    return self.coingecko_client.get_aoa_usd_price()  File "/app/safe_transaction_service/tokens/clients/coingecko_client.py", line 128, in get_aoa_usd_price    return self.get_price("aurora")  File "/app/safe_transaction_service/tokens/clients/coingecko_client.py", line 91, in get_price    return self._get_price(url, name)  File "/app/safe_transaction_service/tokens/clients/coingecko_client.py", line 78, in _get_price    raise CannotGetPrice(safe_transaction_service.tokens.clients.exceptions.CannotGetPrice: Cannot get price from Coingecko for token=aurora
```

--- a/safe_transaction_service/tokens/tests/test_views.py
+++ b/safe_transaction_service/tokens/tests/test_views.py
@@ -13,6 +13,7 @@ from rest_framework.test import APITestCase
 from gnosis.eth.ethereum_client import Erc20Info, Erc20Manager, InvalidERC20Info
 from gnosis.safe.tests.safe_test_case import SafeTestCaseMixin
 
+from ..clients import CannotGetPrice
 from ..models import Token
 from ..services import PriceService
 from ..services.price_service import FiatCode, FiatPriceWithTimestamp
@@ -191,3 +192,19 @@ class TestTokenViews(SafeTestCaseMixin, APITestCase):
         self.assertEqual(response.data["fiat_code"], "USD")
         self.assertEqual(response.data["fiat_price"], "321.2")
         self.assertTrue(response.data["timestamp"])
+
+    @mock.patch.object(
+        PriceService,
+        "get_native_coin_usd_price",
+        side_effect=CannotGetPrice(),
+    )
+    def test_token_price_view_error(self, get_native_coin_usd_price_mock: MagicMock):
+        token_address = "0x0000000000000000000000000000000000000000"
+
+        response = self.client.get(
+            reverse("v1:tokens:price-usd", args=(token_address,))
+        )
+
+        self.assertEqual(response.status_code, status.HTTP_503_SERVICE_UNAVAILABLE)
+        self.assertEqual(response.data["message"], "Price retrieval failed")
+        self.assertEqual(response.data["arguments"], [token_address])

--- a/safe_transaction_service/tokens/views.py
+++ b/safe_transaction_service/tokens/views.py
@@ -13,6 +13,7 @@ from gnosis.eth.constants import NULL_ADDRESS
 from gnosis.eth.utils import fast_is_checksum_address
 
 from . import filters, serializers
+from .clients import CannotGetPrice
 from .models import Token
 from .services import PriceServiceProvider
 from .tasks import get_token_info_from_blockchain_task
@@ -78,25 +79,34 @@ class TokenPriceView(GenericAPIView):
                     "arguments": [address],
                 },
             )
+        try:
+            price_service = PriceServiceProvider()
+            if address == NULL_ADDRESS:
+                data = {
+                    "fiat_code": "USD",
+                    "fiat_price": str(price_service.get_native_coin_usd_price()),
+                    "timestamp": timezone.now(),
+                }
+            else:
+                token = self.get_object()  # Raises 404 if not found
+                fiat_price_with_timestamp = next(
+                    price_service.get_cached_usd_values([token.get_price_address()])
+                )
+                data = {
+                    "fiat_code": fiat_price_with_timestamp.fiat_code.name,
+                    "fiat_price": str(fiat_price_with_timestamp.fiat_price),
+                    "timestamp": fiat_price_with_timestamp.timestamp,
+                }
+            serializer = self.get_serializer(data=data)
+            assert serializer.is_valid()
+            return Response(status=status.HTTP_200_OK, data=serializer.data)
 
-        price_service = PriceServiceProvider()
-        if address == NULL_ADDRESS:
-            data = {
-                "fiat_code": "USD",
-                "fiat_price": str(price_service.get_native_coin_usd_price()),
-                "timestamp": timezone.now(),
-            }
-        else:
-            token = self.get_object()  # Raises 404 if not found
-            fiat_price_with_timestamp = next(
-                price_service.get_cached_usd_values([token.get_price_address()])
+        except CannotGetPrice:
+            return Response(
+                status=status.HTTP_503_SERVICE_UNAVAILABLE,
+                data={
+                    "code": 1,
+                    "message": "Price retrieval failed",
+                    "arguments": [address],
+                },
             )
-            data = {
-                "fiat_code": fiat_price_with_timestamp.fiat_code.name,
-                "fiat_price": str(fiat_price_with_timestamp.fiat_price),
-                "timestamp": fiat_price_with_timestamp.timestamp,
-            }
-
-        serializer = self.get_serializer(data=data)
-        assert serializer.is_valid()
-        return Response(status=status.HTTP_200_OK, data=serializer.data)

--- a/safe_transaction_service/tokens/views.py
+++ b/safe_transaction_service/tokens/views.py
@@ -105,7 +105,7 @@ class TokenPriceView(GenericAPIView):
             return Response(
                 status=status.HTTP_503_SERVICE_UNAVAILABLE,
                 data={
-                    "code": 1,
+                    "code": 10,
                     "message": "Price retrieval failed",
                     "arguments": [address],
                 },

